xarray/tests/test_dataset.py

```diff
@@ -3906,6 +3906,12 @@ def test_filter_by_attrs(self):
         for var in new_ds.data_vars:
             assert new_ds[var].height == '10 m'
 
+        # Test return empty Dataset due to conflicting filters
+        new_ds = ds.filter_by_attrs(
+                standard_name='convective_precipitation_flux',
+                height='0 m') 
+        assert not bool(new_ds.data_vars)
+
```

Comment:
Can you add another test case with a positive example, e.g., one that selects the matching data variable that satisfies both of two different criteria?